backend_new/src/services/listing.service.ts

```diff
@@ -0,0 +1,416 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import {
+  LanguagesEnum,
+  Prisma,
+  ReviewOrderTypeEnum,
+  Units,
+  UnitsSummary,
+} from '@prisma/client';
+import { ListingsQueryParams } from '../dtos/listings/listings-query-params.dto';
+import {
+  calculateSkip,
+  calculateTake,
+  shouldPaginate,
+} from '../utilities/pagination-helpers';
+import { buildOrderBy } from '../utilities/build-order-by';
+import { ListingFilterParams } from '../dtos/listings/listings-filter-params.dto';
+import { ListingFilterKeys } from '../enums/listings/filter-key-enum';
+import { buildFilter } from '../utilities/build-filter';
+import { SummaryOfUnits } from '../dtos/units/summary-of-units.dto';
+
+export type getListingsArgs = {
+  skip: number;
+  take: number;
+  orderBy: any;
+  where: Prisma.ListingsWhereInput;
+};
+
+export const FullListing = Prisma.validator<Prisma.ListingsArgs>()({
```

Comment:
I've done very little research on this so take it with a grain of salt. But did you look into this blog post that says for most cases you can get away without using this validator if you are using typescript. https://www.prisma.io/blog/satisfies-operator-ur8ys8ccq7zb

Reply:
because of nestjs needing us to provide DTOs so the input/output of the backend can be typesafe I ended up switching these to DTOs 

that is really cool though and almost defo will come in hand when moving over the csv export files!